src/main/java/org/apache/maven/plugins/clean/Cleaner.java

```diff
@@ -309,24 +315,23 @@ private int delete(Path file, boolean failOnError, boolean retryOnError) throws
                 for (int delay : delays) {
                     try {
                         Thread.sleep(delay);
-                    } catch (InterruptedException e2) {
-                        exception.addSuppressed(e2);
+                    } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
```

Comment:
I'm confused. If the thread has already thrown an interrupted exception, why are you interrupting it again? Are these two different threads?

Reply:
As we don't re-throw InterruptedException here we should set interrupted flag agin

http://www.javapractices.com/topic/TopicAction.do?Id=251

Reply:
I'm  not sure that's what that says. It suggests resetting the interrupted flag if you don;t handle the exception, but here it's handled by throwing an IOException.

And now that I look at this, is that even right? We're sleeping between retries, the sleep is interrupted, so we cancel all the retries and throw an IOException instead of returning. I guess that makes sense, but at this point the interrupt has been handled so it doesn't seem like the current thread should still have its interrupt flag set.

Reply:
ok remove it.